swayfmt/src/comments.rs

```diff
@@ -32,24 +86,136 @@ pub fn maybe_write_comments_from_map(
             writeln!(formatted_code)?;
         }
 
-        for comment in comments_iter {
-            // Write comments on a newline (for now). New behavior might be required
-            // to support trailing comments.
-            writeln!(
-                formatted_code,
-                "{}{}",
-                formatter.shape.indent.to_string(&formatter.config)?,
-                comment.span().as_str(),
-            )?;
+        while let Some(comment) = comments_iter.next() {
+            let newlines = collect_newlines_after_comment(&formatter.comments_context, comment);
+
+            if formatted_code.trim_end().ends_with(&[']', ';']) {
+                match comment.comment_kind {
+                    CommentKind::Newlined => {
+                        write!(
+                            formatted_code,
+                            "{}{}{}",
+                            formatter.shape.indent.to_string(&formatter.config)?,
+                            comment.span().as_str(),
+                            newlines
+                        )?;
+                    }
+                    CommentKind::Trailing => {
+                        write_trailing_comment(formatted_code, comment)?;
+                        if comments_iter.peek().is_none() {
+                            write!(
```

Comment:
Wouldn't this write `formatted_code` twice since `write_trailing_comment` also writes it? Could you add a comment here?

Reply:
I'm actually going to fix this in a follow up PR - this behaviour probably isn't what we want, its an indirection to indent within write_comments(), but we should we do it one-level-up instead.